Lib/ast.py

```diff
@@ -812,8 +822,17 @@ def visit_Module(self, node):
             ignore.lineno: f"ignore{ignore.tag}"
             for ignore in node.type_ignores
         }
-        self._write_docstring_and_traverse_body(node)
-        self._type_ignores.clear()
+        try:
+            self._write_docstring_and_traverse_body(node)
+        finally:
+            self._type_ignores.clear()
```

Comment:
Another question is whether the mapping precedences should be cleared in visit() as well.